components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -185,6 +204,30 @@ void PageContentExtractor::ExtractPageContent(
                      weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
 }
 
+void PageContentExtractor::OnInterceptedPageContentChanged(
+    std::unique_ptr<AIChatResourceSnifferThrottleDelegate::InterceptedContent>
+        content_update) {
+  DCHECK_EQ(content_update->type,
+            AIChatResourceSnifferThrottleDelegate::InterceptedContentType::
+                kYouTubeMetadataString)
+      << " - unexpected content type";
+  DCHECK(!content_update->content.empty());
+
+  // Store the new content for later, when we're asked for it, so that we
+  // don't have to do any parsing or fetching when there's no active
+  // conversation.
+  intercepted_content_ = std::move(content_update);
+  // Let the host know that new content was received so that it may record a
+  // "page" change.
+  mojo::AssociatedRemote<mojom::PageContentExtractorHost> host;
+  render_frame()->GetRemoteAssociatedInterfaces()->GetInterface(&host);
+  if (host.is_bound()) {
```

Comment:
why do you need to check if it's bound? `GetInterface` gurantees a bind, right? it's internally calling `BindNewEndpointAndPassReceiver` on a remote.

Reply:
removed the condition